### ISSUE TITLE
Pass $model in extraColumns anonymous function

### ISSUE THREAD

**Issue by kirc0de** [yii2tech/ar-linkmany#5]
I need to get id of related model to set the value of my ExtraColumn.

Like this:

```
'extraColumns' => [
    'option_id' => function($model) {
        return isset($this->optionsIds[$model->id]) ? $this->optionsIds[$model->id] : null;
    }
]
```

I will create pull request for it.

**Comment by klimov-paul** [comment]
This make absolutely no sense.
It is much more batter to refer `extraColumns` callback to the model intrenal method instead:

``` php
'extraColumns' => [$this, 'extraColumnsCallback'],
// ...
public function extraColumnsCallback()
{
}
```

Alternativly you can pass `$model` via `use` construction:

```
'extraColumns' => function () use ($this) {},
```

**Comment by kirc0de** [comment]
I think, I don't understand you.

I have Bid model.

Bid has many services, it's many-to-many relation.
Also, i have option_id field in bids_services table (extra-column).

So, in behavior I need to get current Service Model (it's id), but not current Bid model ($this).

```
'linkServicesBehavior' => [
                'class' => LinkManyBehavior::className(),
                'relation' => 'services',
                'relationReferenceAttribute' => 'servicesIds',
                'extraColumns' => [
                    'option_id' => function($currentServiceModel) {
                        // $currentServiceModel <- i need it here, not Bid model
                    }
                ]
            ],
```

**Comment by kirc0de** [comment]
Can you look throught the pull request for understanding the problem?

**Comment by klimov-paul** [comment]
Resolved by commit ffcbf36
